tests/test_cases/test_cocotb/test_handle.py

```diff
@@ -70,6 +71,178 @@ async def test_delayed_assignment_still_errors(dut):
         dut.stream_in_int <= []
 
 
+async def int_values_test(signal, values):
+    """
+    Test integer access to a signal
+    """
+    log = logging.getLogger("cocotb.test")
+    for val in values:
+        signal <= val
```

Comment:
Interestingly, if this line is changed to ```signal = val```, running a test will result in following error:
```
#     20.00ns ERROR    Test Failed: test_int_8bit (result was AttributeError)
#
#                      Traceback (most recent call last):
#
#                        File "C:/cocotb/tests/test_cases/test_cocotb/test_handle.py", line 117, in test_int_8bit
#
#                          await int_values_test(dut.stream_in_data, values)
#
#                        File "C:/cocotb/tests/test_cases/test_cocotb/test_handle.py", line 84, in int_values_test
#
#                          got = signal.value.signed_integer
#
#                      AttributeError: 'int' object has no attribute 'value'
```
Is this expected?

Reply:
Yes that's expected. That is just binding the variable `signal` to the value of `val`, not setting the value of the signal.

Reply:
Ok, thanks for confirming